packages/@react-aria/i18n/src/utils.ts

```diff
@@ -50,14 +50,12 @@ export function numberFormatSignDisplayPolyfill(numberFormat: Intl.NumberFormat,
     }
 
     if (needsPositiveSign) {
-      let negative = numberFormat.format(-num);
-      let noSign = numberFormat.format(num);
-      // ignore RTL/LTR marker character
-      let minus = negative.replace(noSign, '').replace(/\u200e|\u061C/, '');
-      if ([...minus].length !== 1) {
+      let minus = numberFormat.formatToParts(-num).find(part => part.type === 'minusSign').value;
+      if (minus.length === 0) {
         console.warn('@react-aria/i18n polyfill for NumberFormat signDisplay: Unsupported case');
       }
-      let positive = negative.replace(noSign, '!!!').replace(minus, '+').replace('!!!', noSign);
+      let negative = numberFormat.format(-num);
+      let positive = negative.replace(minus, '+');
```

Comment:
Just wanna make sure that I understand this code block above: essentially this adds a '+' sign in from of the number except if the current numberFormat doesn't allow displaying signs?

Reply:
Looks good to me otherwise

Reply:
exceptZero is an option not supported everywhere, so what this does is, if it's a positive number, gets the formatted negative number equivalent and the actual minusSign character. Then it replaces the minus sign character with the positive sign.